src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -589,30 +610,30 @@ public void updateSyntheticClassForSuperCall(Expression expr) {
           this.parentClass,
           methodAndReturnType.getOrDefault(expr.asMethodCallExpr().getNameAsString(), ""));
     } else {
-      updateUnsolvedClassWithVariables(
+      updateUnsolvedClassWithFields(
           expr.asFieldAccessExpr().getNameAsString(),
           parentClass,
           classAndPackageMap.getOrDefault(parentClass, this.currentPackage));
     }
   }
 
   /**
-   * This method will add a new variable declaration to a synthetic class. This class is mainly used
-   * for unsolved parent class. The declaration of the variable in the parent class will be the same
-   * as the declaration in the child class since Specimin does not have access to much information.
-   * If the variable is not found in the child class, Specimin will create a synthetic class to be
-   * the type of that variable.
+   * This method will add a new field declaration to a synthetic class. This method is mainly used
+   * for unsolved parent class. The declaration of the field in the parent class will be the same as
```

Comment:
Throughout, it might be better to use the term "superclass" rather than "parent class". "Superclass" is a more technical word, but I think it is correct to describe what you mean by "parent class" and so should be preferred, rather than using a new term.

Reply:
I'm a bit confused. As I understand, I should keep "parent class" but explain in the document what I mean by "parent class". So I should add a line to this Javadoc saying that a parent class is the super class of the currently visited class. Is that correct, Professor?

Reply:
> I'm a bit confused. As I understand, I should keep "parent class" but explain in the document what I mean by "parent class".

It depends on what exactly you mean by "parent class":
* if "parent class" means exactly the same as "superclass", you should remove all mentions of "parent class" and say "superclass" instead, because "superclass" is the standard term
* if "parent class" means something slightly different than "superclass", you should explain what you mean in the Javadoc

I'm not sure which of those is true, which I think is why my comment was confusing - I was trying to cover both possibilities.

Reply:
Thank you for the explanation.